_posts/en/newsletters/2020-09-16-newsletter.md

```diff
@@ -0,0 +1,218 @@
+  maintainers of various Bitcoin and Bitcoin-derived full nodes.  The
+  attack consisted of `inv` (inventory) messages Bitcoin nodes use to
+  notify their peers of the hash of new transactions (or other data).
+  Normally, when a peer receives an inventory, it checks to see if it
+  already has a transaction with that hash and then requests full copies
+  of any previously-unseen transactions.  In the attack, an attacker would flood the
+  victim with an excessive number of `inv` messages, each containing
+  nearly the maximum allowed number of transaction hashes.  When too
+  many of these inventories were queued, the victim's node would run out
+  of memory and crash.
+
+    Several of the affected programs had code that limited the number of
+    inventories that should've been queued, but the researchers were
+    able to circumvent those protections.  As a bug that could only crash
+    nodes, it could not be used to directly steal from affected nodes.
+    However, it could be used to execute [eclipse attacks][topic eclipse
+    attacks] that could ultimately steal money---although there's no
+    indication the attack was ever used in public.
+
+    All users are encouraged to upgrade to the latest released version
+    of their preferred full node software or to review the list of
+    vulnerable and patched versions on the [website][invdos site].
+
+- **Stealing onchain fees from LN HTLCs:** Antoine Riard [posted][riard
+  post] to the Lightning-Dev mailing list a potential vulnerability in a
+  recent update to the LN specification (see [Newsletter #112][news112
+  bolts688]). When LN payments are made according to the updated
+  specification, each remote party includes a [flag][sighash] in the signature for
+  their half of the 2-of-2 multisig script.  The flag, called
+  `SIGHASH_SINGLE|SIGHASH_ANYONECANPAY`, allows the payment (called an
+  HTLC) to be placed into a transaction containing other inputs and
+  outputs.  This is designed to allow multiple HTLCs  to be bundled
+  together into a transaction that also contains an extra input that
+  pays any additional transaction fees and an optional extra output that
+  collects any money left over from paying those extra fees.  This makes
+  it possible to wait until those HTLCs need to be confirmed before
+  choosing what feerate to pay.
+
+    {:.center}
+    ![FIXME](/img/posts/2020-09-htlc-fee-bumping.dot.png)
+
+    However, Riard notes that the specification's earlier mechanism of
+    committing to variable feerates for HTLCs is still available.  This
+    can allow an attacker to commit to a high fee for multiple HTLCs
+    and then construct a transaction that combines those HTLCs with just
+    an additional output that claims those fees.  Riard's email
+    describes how to use this basic technique in a more complex attack
+    to maximize the amount stolen.
+
+    {:.center}
+    ![FIXME](/img/posts/2020-09-htlc-fee-stealing.dot.png)
```

Comment:
Again, I think this could be slightly confusing for people who aren't very familiar with Bitcoin's fee mechanism. It shows implicit fees for the first two input-output pairs, but there aren't actually any fees paid because of the additional output.